src/activity_item_handling.cpp

```diff
@@ -7,266 +9,383 @@
+    }
+
+    if( fallen_count > 0 ) {
+        add_msg( m_warning, _( "The trunk is full, so some items fell to the %2$s." ), ter_name.c_str() );
+    }
 }
 
-static tripoint get_item_pointers_from_activity(
-    std::list<item *> &selected_items, std::list<int> &item_quantities,
-    std::list<item *> &selected_worn_items, std::list<int> &worn_item_quantities )
+void stash_on_pet( const std::vector<item> &items, monster &pet )
 {
-    // Drop activity has indices of items in inventory and quantities of same.
-    // Indices and quantities alternate on the list.
-    // First iterate over the indices and quantities, retrieving item references.
-    for( size_t index = 0; index < g->u.activity.values.size(); index += 2 ) {
-        const int position = g->u.activity.values[index];
-        const int quantity = g->u.activity.values[index + 1];
-        const bool is_worn = position < -1;
-        const bool is_weapon = position == -1;
-        if( is_worn ) {
-            item &armor = g->u.i_at( position );
-            assert( !armor.is_null() );
-            selected_worn_items.push_back( &armor );
-            worn_item_quantities.push_back( quantity );
-        } else  if( is_weapon ) {
-            selected_items.push_back( &g->u.weapon );
-            item_quantities.push_back( quantity );
-        } else {
-            // We MUST have pointers to these items, and this is the only way I can see to get them.
-            std::list<item> &stack = ( std::list<item> & )g->u.inv.const_stack( position );
-            int items_dropped = 0;
-            for( auto &elem : stack ) {
-                selected_items.push_back( &elem );
-                if( elem.count_by_charges() ) {
-                    const int qty_to_drop = std::min( quantity - items_dropped, ( int )elem.charges );
-                    item_quantities.push_back( qty_to_drop );
-                    items_dropped += qty_to_drop;
-                } else {
-                    item_quantities.push_back( 1 );
-                    items_dropped++;
-                }
-                if( items_dropped >= quantity ) {
-                    break;
-                }
+    if( pet.inv.empty() ) {
+        return;
+    }
+    for( auto &it : items ) {
+        int max_volume = pet.inv.front().get_storage();
+        int max_weight = pet.weight_capacity();
+
+        for( const auto &it : pet.inv ) {
+            max_volume -= it.volume();
+            max_weight -= it.weight();
+        }
+
+        const bool too_heavy = it.weight() > max_weight;
+        const bool too_big = it.volume() > max_volume;
+
+        pet.add_effect( effect_controlled, 5 );
+
+        if( too_heavy || too_big ) {
+            const std::string it_name = it.display_name();
+            const std::string ter_name = g->m.name( pet.pos() );
+
+            if( too_big ) {
+                add_msg( m_bad, _( "%s did not fit and fell to the %1$s." ), it_name.c_str(), ter_name.c_str() );
```

Comment:
`it_name` will be printed twice (%s and %1$s), but `ter_name` will not be printed at all. Same with the message below.